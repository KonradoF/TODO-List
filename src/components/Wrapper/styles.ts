import EStyleSheet from 'react-native-extended-stylesheet';

export const styles = EStyleSheet.create({
  wrapper: {
    paddingHorizontal: '$wrapperPaddingHorizontal',
    backgroundColor: '$white',
    flex: 1,
  },
  scrollWrapper: {flexGrow: 1, paddingBottom: '$wrapperPaddingBottom'},
  noSpacing: {paddingHorizontal: 0},
  spacingTop: {paddingTop: '$wrapperPaddingTop'},
});
