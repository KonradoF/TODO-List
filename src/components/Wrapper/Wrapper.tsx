import React, {ReactElement, ReactNode} from 'react';
import {View, ScrollView, ViewStyle} from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import {styles} from './styles';

type Props = {
  children?: ReactNode;
  noScroll?: boolean;
  spacingTop?: boolean;
  style?: ViewStyle;
  noSpacing?: boolean;
  keyboardShouldPersistTaps?: 'handled' | 'always';
  bounces?: boolean;
  bgColor?: boolean;
};
export const Wrapper = ({
  noSpacing,
  children,
  noScroll = false,
  spacingTop,
  style,
  keyboardShouldPersistTaps,
  bounces = false,
  bgColor = false,
}: Props): ReactElement => {
  if (noScroll) {
    return (
      <View
        style={EStyleSheet.flatten([
          styles.wrapper,
          noSpacing && styles.noSpacing,
          spacingTop && styles.spacingTop,
          style,
        ])}>
        {children}
      </View>
    );
  } else {
    return (
      <ScrollView
        keyboardShouldPersistTaps={keyboardShouldPersistTaps}
        bounces={bounces}
        showsVerticalScrollIndicator={false}
        contentContainerStyle={EStyleSheet.flatten([
          styles.scrollWrapper,
          spacingTop && styles.spacingTop,
          style,
        ])}
        style={EStyleSheet.flatten([
          styles.wrapper,
          noSpacing && styles.noSpacing,
          bgColor && {backgroundColor: 'transparent'},
        ])}>
        {children}
      </ScrollView>
    );
  }
};
