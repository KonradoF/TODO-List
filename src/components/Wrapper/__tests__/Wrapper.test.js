import React from 'react';
import {shallow} from 'enzyme';
import {Text} from 'react-native-elements';
import {Wrapper} from 'src/components/Wrapper/Wrapper';

describe('Wrapper', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<Wrapper />);
      expect(component).toMatchSnapshot();
    });
    it('should match to snapshot no scroll ', () => {
      const component = shallow(<Wrapper noScroll />);
      expect(component).toMatchSnapshot();
    });
    it('should match to snapshot spacing top ', () => {
      const component = shallow(<Wrapper spacingTop />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('Check render props', () => {
    let component;
    const text = 'Test';

    beforeEach(() => {
      component = shallow(
        <Wrapper>
          <Text>{text}</Text>
        </Wrapper>,
      );
      jest.clearAllMocks();
    });

    it('should render correct children', () => {
      const labelElement = component.find({children: text});
      expect(labelElement).toHaveLength(1);
    });
  });
  describe('Chceck render props no scroll', () => {
    let component;
    const text = 'Test';

    beforeEach(() => {
      component = shallow(
        <Wrapper noScroll>
          <Text>{text}</Text>
        </Wrapper>,
      );
      jest.clearAllMocks();
    });

    it('should render correct children', () => {
      const labelElement = component.find({children: text});
      expect(labelElement).toHaveLength(1);
    });
  });
});
