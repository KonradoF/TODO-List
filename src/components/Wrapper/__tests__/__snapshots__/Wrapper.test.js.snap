// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper Rendering should match to snapshot 1`] = `
<ScrollView
  bounces={false}
  contentContainerStyle={
    Object {
      "flexGrow": 1,
      "paddingBottom": 20,
    }
  }
  showsVerticalScrollIndicator={false}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "paddingHorizontal": 48,
    }
  }
/>
`;

exports[`Wrapper Rendering should match to snapshot no scroll  1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "paddingHorizontal": 48,
    }
  }
/>
`;

exports[`Wrapper Rendering should match to snapshot spacing top  1`] = `
<ScrollView
  bounces={false}
  contentContainerStyle={
    Object {
      "flexGrow": 1,
      "paddingBottom": 20,
      "paddingTop": 32,
    }
  }
  showsVerticalScrollIndicator={false}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "paddingHorizontal": 48,
    }
  }
/>
`;
